pkg/segment/metadata/segmentmicroindex.go

```diff
@@ -287,14 +306,148 @@ func (sm *SegmentMicroIndex) GetColumns() map[string]bool {
 	return retVal
 }
 
-func (sm *SegmentMicroIndex) GetAllColumnsRecSize() map[string]uint32 {
+func (sm *SegmentMicroIndex) getAllColumnsRecSize() map[string]uint32 {
 	retVal := make(map[string]uint32, len(sm.ColumnNames))
 	for colName, colSizeInfo := range sm.ColumnNames {
 		retVal[colName] = colSizeInfo.ConsistentCvalSize
 	}
 	return retVal
 }
 
-func (sm *SegmentMicroIndex) GetRecordCount() uint32 {
+func (sm *SegmentMicroIndex) getRecordCount() uint32 {
 	return uint32(sm.SegMeta.RecordCount)
 }
+
+func GetLoadSsm(segkey string, qid uint64) (*SegmentMicroIndex, int64, error) {
+
+	smi, exists := getMicroIndex(segkey)
+	if !exists {
+		log.Errorf("qid=%d, Seg  %+v does not exist in block meta, but existed in time filtering", qid, segkey)
+		return nil, 0, fmt.Errorf("seg file %+v does not exist in block meta, but existed in time filtering", segkey)
+	}
```

Comment:
nit: you can use `utils.TeeErrorf()` to log and return an error. But you can keep what you have if you want